### PR TITLE
add client_design_doc.md

### PR DESCRIPTION
**Pull request by tonyyang-svail**

### CODE REVIEW

File: doc/client_design_doc.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+SQLFlow Client implements a minimum client that issues queries to and fetch results/logs from [sqlflowserver](https://github.com/wangkuiyi/sqlflowserver).
+
+In `pysqlflow`, the client will be wrapped in the magic command where user directly writes SQL statements.
+
+## Example
+
+```python
+import sqlflow
+
+client = sqlflow.Client(address='localhost:50051')
```
Comment (ktong) [inline doc/client_design_doc.md]: Is base_url or server_url better than address as address is too generic.

File: doc/client_design_doc.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# Regular SQL
+response = client.query('select * from table')
+for dataframe in response:
+    print(dataframe) # {'y': [4, 5, 6], 'x': [1, 2, 3]}
+
+# ML SQL
+response = client.query('select ... train ...')
```
Comment (ktong) [inline doc/client_design_doc.md]: I am thingking adding client.train(...) as wrapper of execute trainning. So does client.predict(...).

Comment (ktong) [reply doc/client_design_doc.md]: There is no return value for log as implementation log messages from server using logging

File: doc/client_design_doc.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+class Response(object):
+    def __init__(self, response):
+        self._response = response
+    
+    def _decode_protobuf(self, proto):
+        ...
+
+    def __iter__(self):
+        for res_proto in self.response:
+            return self._decode_protobuf(res_proto)
```
Comment (ktong) [inline doc/client_design_doc.md]: Suggest adding to_dataframe() to convert result to pandas dataframe when pandas exists

File: doc/client_design_doc.md (hunk 1 above)
Comment (tonyyang-svail) [reply doc/client_design_doc.md]: Done.

File: doc/client_design_doc.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+client.query('select ... train ...')
+```
+
+## Implementation
+
+`sqlflow.Client` uses grpc to contact the `sqlflowserver`. The service protocol is defined at [sqlflow.proto](https://github.com/wangkuiyi/sqlflowserver/blob/develop/sqlflow.proto).
+
+`sqlflow.Client.__init__` establishes a grpc stub/channel based on `server_url`.
+
+`sqlflow.Client.query` takes a sql statement
```
Comment (ktong) [inline doc/client_design_doc.md]: May we change query to execute or run as this sql is not just for query?

File: doc/client_design_doc.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        channel = grpc.insecure_channel(host)
+        self._stub = sqlflow_pb2_grpc.SQLFlowStub(channel)
+
+    def _decode_protobuf(self, proto):
+        ...
+
+    def query(self, query):
+        for res in self._stub.Run(sqlflow_pb2.RunRequest(sql=operation)):
+            if res.is_message():
+                print(res)
```
Comment (ktong) [inline doc/client_design_doc.md]: log, not print

Comment (tonyyang-svail) [reply doc/client_design_doc.md]: Done.

File: doc/client_design_doc.md (hunk 4 above)
Comment (tonyyang-svail) [reply doc/client_design_doc.md]: good idea.

**Pull request merged**